src/main/java/net/openhft/chronicle/wire/FieldInfo.java

```diff
@@ -145,5 +200,34 @@ public interface FieldInfo {
      */
     Class<?> genericType(int index);
 
+    /**
+     * Copy the field from the source object to the destination
+     * <p>
+     * Note: this is not a deep copy, objects will be copied by reference
+     *
+     * @param source      The object to copy the field value from
+     * @param destination The object to copy the field value to
+     */
+    default void copy(Object source, Object destination) {
```

Comment:
do we still need this?

Reply:
We don't need it, but it's probably nice to have. To copy a field value from one instance to another you can use set(dest, get(source)), which will use reflection and box/unbox in the event it's a primitive field, or you can write an if/else if/else if... to use the correct primitive setter/getter, but if we have this method each field type can implement it's own efficient way of doing it.
It's only useful in one place (in Wire) that I can see (dynamic enums), so you could argue it's unnecessary, but it's not a big cost to implement/maintain.

Reply:
Oh and if you add a method to an interface binary-compatibility-checker makes you add a default implementation too. I just put the box/unbox one in as the default